### PR TITLE
Fix APIMANAGER-5805

### PR DESCRIPTION
**Pull request by abeykoon**
This PR fixes the issue of removing brackets around a single element Json
response when response caching is enabled.

Fixes: https://wso2.org/jira/browse/APIMANAGER-5805
Fixes: https://github.com/wso2/product-ei/issues/691

### CODE REVIEW

**Comment by abeykoon** [comment]
The fix is to address issues when caching PIs within SOAP envelope, which is used to indicate arrays in Json when converted to SOAP format.

File: components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java [hunk 1]
```diff
@@ -354,38 +371,66 @@ private boolean processRequestMessage(MessageContext synCtx,
...
+						} catch (XMLStreamException e) {
+							handleException("Error creating response OM from cache : " + cacheKey, synCtx);
+						}
 					}
 
-					if (omSOAPEnv != null) {
-						synCtx.setEnvelope(omSOAPEnv);
+					if (response != null) {
+						// Set the headers of the message
+						if (response.getFirstElement().getLocalName().contains(HEADER)) {
```
Comment (milindaperera) [inline components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java]: shall we merge these two if statements (line No 404 and 406) to one statement

Comment (abeykoon) [inline components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java]: We have some other parts outside the second if statement, but within first statement. So maybe better to keep it as it is? 

if (msgCtx.getEnvelope().getBody().getFirstElement() != null) {
	msgCtx.getEnvelope().getBody().getFirstElement().detach();
}						msgCtx.getEnvelope().getBody().addChild(response.getFirstElement().getFirstElement());

Comment (milindaperera) [inline components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediator.java]: oops, I didn't notice that :)
